plugins-mu/hm-stack/class-endpoint-controller.php

```diff
@@ -0,0 +1,238 @@
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_environment' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+			'schema'              => [
+				'$schema'              => 'http://json-schema.org/draft-04/schema#',
+				'title'                => 'environment-data',
+				'type'                 => 'object',
+				'properties'           => [
+					'environmentData' => [
+						'description'  => esc_html__( 'Data about the current server environment', 'my-textdomain' ),
+						'type'         => 'object',
+						'context'      => array( 'view' ),
+						'readonly'     => true,
+					],
+					'gitData' => [
+						'description'  => esc_html__( 'Data about the current state of git on the site.', 'my-textdomain' ),
+						'type'         => 'object',
+						'context'      => array( 'view' ),
+						'readonly'     => true,
+					],
+				],
+			]
+		] );
+
+		// Fetch all pull requests against this site.
+		register_rest_route( $namespace, 'pull-requests', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_pull_requests' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch average page generation time for this site.
+		register_rest_route( $namespace, 'page-generation', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_page_generation_time' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+	}
+
+	/**
+	 * Validate whether the current user has appropriate capabilities to fetch HM Stack data or not.
+	 *
+	 * @return bool
+	 */
+	public function permissions_check() : bool {
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Make a human-readable error code from various Hm Stack responses.
+	 *
+	 * @param $error WP_Error Error to parse.
+	 * @return WP_Error
+	 */
+	private function get_return_error_message( WP_Error $error ) : WP_Error {
+		switch( $error->get_error_message() ) {
+			case 'Authentication Failed':
+				return new WP_Error(
+					'platform.hmstack.api.could_not_authenticate',
+					'Unable to authenticate to retrieve data.',
+					['status' => WP_Http::INTERNAL_SERVER_ERROR]
+				);
+			default :
+				return new WP_Error(
+					'platform.hmstack.api.could_not_connect',
+					'Unable to connect with the HM Stack API. Error: ' . $error->get_error_message(),
+					['status' => WP_Http::INTERNAL_SERVER_ERROR]
+				);
+		}
+	}
+
+	/**
+	 * @todo:: Need more information to fetch this data.
+	 */
+	public function get_bandwidth_usage() {}
+
+	/**
+	 * Send off environmental data about the site such as gitBranch information and PHP version.
+	 *
+	 * @return WP_REST_Response|\WP_Error
+	 */
+	public function get_environment() {
```

Comment:
`get_environment` would typically be the environment name, `get_environment_data()` maybe? But `environment` still seems a bit ambiguous.